src/openvic-simulation/economy/production/ResourceGatheringOperation.cpp

```diff
@@ -105,12 +105,20 @@ fixed_point_t ResourceGatheringOperation::calculate_size_modifier() const {
 
 	ProductionType const& production_type = *production_type_nullable;
 
+	//TODO WVPM split rgo size modifiers based on tech/non-tech
 	fixed_point_t size_modifier = fixed_point_t::_1();
-	if (production_type.is_farm()) {
+	if (production_type.get_is_farm_for_tech()) {
+		size_modifier += location.get_modifier_effect_value_nullcheck(modifier_effect_cache.get_farm_rgo_size_global());
+	}
+	if (production_type.get_is_farm_for_non_tech()) {
 		size_modifier += location.get_modifier_effect_value_nullcheck(modifier_effect_cache.get_farm_rgo_size_global())
 			+ location.get_modifier_effect_value_nullcheck(modifier_effect_cache.get_farm_rgo_size_local());
 	}
```

Comment:
```suggestion
	if (production_type.get_is_farm_for_tech()) {
		size_modifier += location.get_modifier_effect_value_nullcheck(modifier_effect_cache.get_farm_rgo_size_global());
	}
	if (production_type.get_is_farm_for_non_tech()) {
		size_modifier += location.get_modifier_effect_value_nullcheck(modifier_effect_cache.get_farm_rgo_size_local());
	}
```
- For the non-tech case you only want to be adding the local modifier, otherwise you'll end up double counting it if both tech and non-tech farm is true.
- Same for the mine modifier effects below.
- Double check exactly which modifier effects are used for `base_size_modifier` in `initialise_rgo_size_multiplier` and for `size_modifier` here, in all 4 main cases: nothing, just `farm = yes`, just `mine = yes`, and both `farm = yes` and `mine = yes`.